src/damage.h

```diff
@@ -90,28 +85,22 @@ struct resistances {
 
 struct projectile {
     damage_instance impact;
-    damage_instance payload;
-    blast_shape aoe_shape;
-    nc_color aoe_color;
-    int aoe_size;
     int speed; // how hard is it to dodge? essentially rolls to-hit, bullets have arbitrarily high values but thrown objects have dodgeable values
-    bool drops; // does it drop ammo units?
-    bool wide; // a shot that "covers" the target, e.g. a shotgun blast or flamethrower napalm
 
-    // TODO: things below here are here temporarily until we finish those
-    // systems
     std::set<std::string> proj_effects;
-    itype *ammo; // projectile's item that gets spawned at impact location, e.g. thrown weapons/bolts
-
-    projectile() :
-        aoe_shape(BS_NONE),
-        aoe_color(c_red),
-        aoe_size(0),
-        speed(0),
-        drops(false),
-        wide(false),
-        ammo(NULL)
-    { }
+    std::unique_ptr<item> drop; // Actual item used (to drop contents etc.). Null in case of bullets (they aren't "made of cartridges")
+
+    projectile();
+    projectile( const projectile& );
+    projectile& operator=( const projectile& );
```

Comment:
Assignment happens in the code, though. For example, bouncing projectiles and shrapnel.